### DISCUSSION TITLE
Archive Schema

### DISCUSSION THREAD

**Discussion by cappelaere** [opendcs/opendcs#162] [Q&A]
Data is stored in the archive in XML.  Is there a schema defined somewhere?
So far I have found two record types but not sure if there are potentially more...
There is one type that escapes me:
```
<DcpMsg platformid="xxxxxxxxxxxxxxxxxxxxxxx">
  <XmitTime>...</XmitTime>...
  <MOMSM>...</MOMSM>
  <CDR_Reference>...</CDR_Reference>
  <SessionStatus>...</SessionStatus>
  <LocalRecvTime>...</LocalRecvTime>
  <BinaryMsg>...</BinaryMsg>
</DcpMsg>
```
How to I retrieve the metadata for that platform? (like agency, location...)
I cannot find that platformid in the PDTS_COMPRESSED.txt file
Thanks.
Pat.

**Comment by MikeNeilson** [comment]
Unfortunately no, other than the code. 

However, for the most part the entries in that file are just what's in https://github.com/opendcs/opendcs/blob/master/src/main/java/lrgs/common/DcpMsg.java.

As for meta data, the PDTS_compressed.txt may just be old, and will only include GOES data, not something like irridium. The LRGS archive only cares from which defined data source (GOES,HRIT, Iridium, etc) and when, basically.

Further meta data for a local configuration would be manually entered into the DECODES database, that can be XML or one of the 3 SQL implementations.

**Reply by MikeNeilson** [reply]
I'm not opposed to changing that behavior long term, having more meta data would be good, but I don't think it would work well as a lot of the collection LRGS does is pretty direct from simple sources, like HRIT, that don't embed the information.

Food for thought for a future service though, let the downstream requestor ask and if defined for a given platform, provide it. But we don't currently have a good way to handle that.

**Comment by cappelaere** [comment]
Are you saying that this record might be an Iridium DCS record ( I guess the flags would tell me)  and we do not have a way to trace the platformId back to an organization?

**Reply by MikeNeilson** [reply]
Correct, or from an EDL file, or a NetDCP directly connected to on the network.

Not with just the LRGS archive itself. Such information was never directly embedded within any of the message types received.
When a user sets up DECODES (the configuration of which is in a separate database) there are field for that information, but it must be entered manually by the user.

**Reply by MikeNeilson** [reply]
that's not to say we can't include it; but it would still have to be manually configured since the sources don't send anything.

**Reply by MikeNeilson** [reply]
I can see that being useful on the NOAA side as the main "DCS" provider. Assuming organizations kept their information up-to-date in DADDS there could be some form of mapping from there.

**Comment by mikecovesw** [comment]
That record is an Iridium message. CDR reference is a unique number for referencing that message on the gateway.

**Reply by cappelaere** [reply]
Thanks Mike and Mike.  It would be nice to have a schema documenting what could go in the archive (or database).  Some of those fields are very cryptic for the non-initiated.  I might have to go back to NOAA to figure out that platformID for Iridium... This would be critical for future capabilities such as data streaming and PubSub.
